core/src/test/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandlerTest.java

```diff
@@ -105,6 +110,28 @@ public void should_fail_if_no_node_available() {
     }
   }
 
+  @Test
+  public void should_fail_if_nodes_unavailable() {
+    RequestHandlerTestHarness.Builder harnessBuilder = RequestHandlerTestHarness.builder();
+    try (RequestHandlerTestHarness harness = harnessBuilder.withEmptyPool(node1).build()) {
+      CompletionStage<AsyncResultSet> resultSetFuture =
+              new CqlRequestHandler(
+                      UNDEFINED_IDEMPOTENCE_STATEMENT,
+                      harness.getSession(),
+                      harness.getContext(),
+                      "test")
+                      .handle();
+      assertThatStage(resultSetFuture)
+              .isFailed(error -> {
+                assertThat(error).isInstanceOf(AllNodesFailedException.class);
```

Comment:
Clearly haven't worked with assertj before, I'd love feedback on how to make this block more readable!

Reply:
That's very well written, congrats :-)

Just as a suggestion, how about we include 2 nodes in this test? Something like:

```
@Test
public void should_fail_if_nodes_unavailable() {
  RequestHandlerTestHarness.Builder harnessBuilder = RequestHandlerTestHarness.builder();
  try (RequestHandlerTestHarness harness =
      harnessBuilder.withEmptyPool(node1).withEmptyPool(node2).build()) {
    CompletionStage<AsyncResultSet> resultSetFuture =
        new CqlRequestHandler(
                UNDEFINED_IDEMPOTENCE_STATEMENT,
                harness.getSession(),
                harness.getContext(),
                "test")
            .handle();
    assertThatStage(resultSetFuture)
        .isFailed(
            error -> {
              assertThat(error).isInstanceOf(AllNodesFailedException.class);
              Map<Node, List<Throwable>> allErrors =
                  ((AllNodesFailedException) error).getAllErrors();
              assertThat(allErrors).hasSize(2);
              assertThat(allErrors)
                  .hasEntrySatisfying(
                      node1,
                      nodeErrors ->
                          assertThat(nodeErrors)
                              .singleElement()
                              .isInstanceOf(NodeUnavailableException.class));
              assertThat(allErrors)
                  .hasEntrySatisfying(
                      node2,
                      nodeErrors ->
                          assertThat(nodeErrors)
                              .singleElement()
                              .isInstanceOf(NodeUnavailableException.class));
            });
  }
}
```

Reply:
thanks, done